### PR TITLE
[CP] New tooling for iOS 17 physical devices (#131865)

### PR DESCRIPTION
**Pull request by vashworth**
Original PR: https://github.com/flutter/flutter/pull/131865

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -3653,6 +3653,16 @@ targets:
         ["devicelab", "ios", "mac"]
       task_name: flutter_gallery_ios__start_up
 
+  - name: Mac_ios flutter_gallery_ios__start_up_xcode_debug
```
Comment (christopherfujino) [inline .ci.yaml:3656]: Do we want these tests in the release branch?

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by vashworth** [comment]
@christopherfujino Added the logging update to this CP (for 3.14) too. Can you re-approve?

File: .ci.yaml (hunk 1 above)
Comment (vashworth) [reply .ci.yaml:3656]: As discussed offline, these tests are in bringup and therefore should be okay

**Review by christopherfujino — APPROVED** [review]
Still LGTM

**Pull request merged**